spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/ReactiveLoadBalancerClientFilter.java

```diff
@@ -119,23 +133,30 @@ public int getOrder() {
 				log.trace("LoadBalancerClientFilter url chosen: " + requestUrl);
 			}
 			exchange.getAttributes().put(GATEWAY_REQUEST_URL_ATTR, requestUrl);
-		}).then(chain.filter(exchange));
+			exchange.getAttributes().put(GATEWAY_LOADBALANCER_RESPONSE_ATTR, response);
+		}).then(chain.filter(exchange))
+				.doOnError(throwable -> supportedLifecycleProcessors.forEach(lifecycle -> lifecycle.onComplete(
+						new CompletionContext<ClientResponse, ServiceInstance>(CompletionContext.Status.FAILED,
+								throwable, exchange.getAttribute(GATEWAY_LOADBALANCER_RESPONSE_ATTR)))))
+				.doOnSuccess(aVoid -> supportedLifecycleProcessors.forEach(
+						lifecycle -> lifecycle.onComplete(new CompletionContext<>(CompletionContext.Status.SUCCESS,
+								exchange.getAttribute(GATEWAY_LOADBALANCER_RESPONSE_ATTR), exchange.getResponse()))));
```

Comment:
Wouldn't it be a problem if the lifecycle processor read the response body here?  You can only read the body once.

Reply:
Yes, thanks for pointing this out. In fact, if they did that, it might be a problem. And we are also similarly passing the entire ClientRequest in Commons in the LB ExchangeFilterFunctions.  So I think in this case, we can either rely on users knowing what they're doing or maybe it would be safer to just pass the information that can be read many times, like status code and headers. WDYT?

Reply:
Will modify it to use a copy/ clone of the response instead.